src/transformState.js

```diff
@@ -5,7 +5,23 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const property of actions) {
+    if (property.type === 'addProperties') {
+      Object.assign(state, property.extraData);
+    }
+
+    if (property.type === 'removeProperties') {
+      property.keysToRemove
+        .forEach(key => delete state[key]);
+    }
+
+    if (property.type === 'clear') {
+      Object.keys(state)
+        .forEach(key => delete state[key]);
+    }
+  }
+
+  return state;
```

Comment:
You need to rewrite the code using a `switch` instead of the `if` statements. As outlined in the checklist.